File: app/src/main/java/org/lynxz/rxjavademo/MainActivity.java
package org.lynxz.rxjavademo;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.btn_basic).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, BasicDemoActivity.class));
            }
        });

        findViewById(R.id.btn_operator).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, OperatorDemoActivity.class));
            }
        });

        findViewById(R.id.btn_error).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, ErrorHandlerActivity.class));
            }
        });
    }


}

File: app/src/main/java/org/lynxz/rxjavademo/MainActivity.kt
package org.lynxz.rxjavademo

import android.content.Intent
import kotlinx.android.synthetic.main.activity_main.*
import org.lynxz.rxjavademo.activity.*
import org.lynxz.rxjavademo.base.BaseActivity

class MainActivity : BaseActivity() {
    override fun getLayoutRes() = R.layout.activity_main
    override fun afterCreate() {
        btn_basic.setOnClickListener { startActivity(Intent(this@MainActivity, BasicDemoActivity::class.java)) }
        btn_operator.setOnClickListener { startActivity(Intent(this@MainActivity, OperatorDemoActivity::class.java)) }
        btn_error.setOnClickListener { startActivity(Intent(this@MainActivity, ErrorHandlerActivity::class.java)) }
        btn_rx_binding.setOnClickListener { startActivity(Intent(this@MainActivity, RxbindingActivity::class.java)) }
        btn_rx_retrofit.setOnClickListener { startActivity(Intent(this@MainActivity, RetrofitDemoActivity::class.java)) }
    }
}
